### PR TITLE
Make master workspace orientation rule dynamic

### PR DESCRIPTION
**Pull request by thejch**
#### Describe your PR, what does it fix/add?
Make master workspace orientation rule dynamic to allow more control. With this if you move a workspace to a different monitor etc it will follow the rules you set for the monitor.

addresses https://github.com/hyprwm/Hyprland/issues/3174
also closes https://github.com/hyprwm/Hyprland/issues/5312

also fixes a special workspace resizing bug

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
no

#### Is it ready for merging, or does it need work?
ready

### CODE REVIEW

**Comment by thejch** [comment]
need to rebase after https://github.com/hyprwm/Hyprland/pull/5340

**Comment by vaxerski** [comment]
tag me when ready

**Comment by thejch** [comment]
> tag me when ready

It's ready

File: src/layout/MasterLayout.cpp [hunk 1]
```diff
@@ -339,6 +339,27 @@ void CHyprMasterLayout::calculateWorkspace(const int& ws) {
...
+    // dynamic workspace rules
+    const auto  WORKSPACERULES = g_pConfigManager->getWorkspaceRulesFor(g_pCompositor->getWorkspaceByID(ws));
+    std::string orientationForWs;
+
+    for (auto& wsRule : WORKSPACERULES) {
+        if (wsRule.layoutopts.contains("orientation"))
+            orientationForWs = wsRule.layoutopts.at("orientation");
+    }
+
+    if (orientationForWs == "top") {
```
Comment (vaxerski) [inline src/layout/MasterLayout.cpp:351]: all this time and we still forget no {} around short ifs? :P

**Review by vaxerski — COMMENTED** [review]
wheres the backdoor

File: src/layout/MasterLayout.cpp (hunk 1 above)
Comment (thejch) [reply src/layout/MasterLayout.cpp]: you got me. i actually hid a backdoor in the {}s.
anyway deleted now

**Review by vaxerski — APPROVED** [review]
lgtm as usual

**Pull request merged**